xoinvader/collision.py

```diff
@@ -66,19 +66,21 @@ def _add(self, collider):
     def update(self):
         """Detect and process all collisions."""
         for pair in self._collisions:
-            c_a = [n for n in self._colliders if n.col_type == pair.a]
-            c_b = [n for n in self._colliders if n.col_type == pair.b]
-            for cldr_a in c_a:
-                for cldr_b in c_b:
-                    maybe_coll_rect = self.check_collision(cldr_a, cldr_b)
-                    if maybe_coll_rect:
-                        callbacks = self._collisions[pair]
-                        for cb in callbacks:
-                            if cb:
-                                cb(maybe_coll_rect)
+            colliders_type_1 = [item for item in self._colliders
+                                if item.col_type == pair.first]
+            colliders_type_2 = [item for item in self._colliders
+                                if item.col_type == pair.second]
+            for collider_1 in colliders_type_1:
+                for collider_2 in colliders_type_2:
+                    maybe_collision_rect = self.check_collision(collider_1,
```

Comment:
I think that `maybe` is excessive here. It's common pattern to check on None some things...
```python
collision_rect = self.check_collision(collider_1, collider_2)
```
From next line is obvious that collision can happen.